### DISCUSSION TITLE
Connection Refused to Servers on AdvantEDGE

### DISCUSSION THREAD

**Discussion by userH1213** [InterDigitalInc/AdvantEDGE#398] [Q&A]
Hello,
 
I'm trying to use my own image, but for some reason I cannot access it through the platform. 
In an attempt to narrow the cause of the problem, I tested accessing the servers from demo1 through the external users. I'm attaching some figures from **demo1 while trying to send some packets.**
![unableAccessingServers](https://user-images.githubusercontent.com/79055649/194737410-299c283a-b826-4838-bc6c-cd408118c7ba.png)
![unableAccessingServers2](https://user-images.githubusercontent.com/79055649/194737417-448c6c8c-6e40-4ee6-afa8-e9f9c4343e41.png)

**Here is the configuration of the server(bs-1) using the image fifo** (I've pushed it to meep-registry)
![configFifo1](https://user-images.githubusercontent.com/79055649/194737498-5a1fbd8f-b4c0-44e7-9dd3-92e13e7d3dc3.png)
![configFifo2](https://user-images.githubusercontent.com/79055649/194737500-fb78eca9-76a4-499e-bbac-75a9d1ac729b.png)
And this is an example of how I'm **configuring the external applications**

![configExternalApp](https://user-images.githubusercontent.com/79055649/194771901-7db8a89e-3e4d-41be-9580-86be6d52215e.png)

In this you can see **the connection is not going through**
![unableAccessingFifo](https://user-images.githubusercontent.com/79055649/194737515-9ceefb5b-50e2-435b-9ff0-3c436a26c243.png)

I've checked the firwall status on my machine and it is inactive.
any help is appreciated.

**Comment by dilallkx** [comment]
Hi @userH1213,

Thank you for using the platform & for reaching out!

From a quick inspection of your above configuration, it seems you are trying to start an iperf3 server with the unsupported '-u' option. Is it possible your server is not running properly in your pod? Try checking the server pod logs or starting a shell in your server pod where you can manually start and test the iperf3 server (note: you can use a tool like k9s to establish a shell in your pod).

I hope this helps you identify the issue.
Regards!

**Comment by userH1213** [comment]
Thank you @dilallkx for your reply. 
I've tested my server in VScode, iperf3 is working fine and supports UDP '-u' ,as you can see in the results here.  Also, on the platform it show that the pod is running. Should I expect different results with k9s ?
![running-fifo-pod](https://user-images.githubusercontent.com/79055649/196007984-ff0ec55b-104d-4788-b34a-49aaa22da4c8.png)



![testing-script](https://user-images.githubusercontent.com/79055649/196007398-3ec7bf1f-ff27-4b78-b0b3-8f2d85564957.png)

![test-scriptr2](https://user-images.githubusercontent.com/79055649/196007406-b9e2d04d-d2d8-44d3-9558-dc3b3685ba66.png)

**Comment by dilallkx** [comment]
Hi @userH1213,

Which version of Iperf3 are you using that supports the -u option for the server? From the [Iperf3 man page](https://www.mankier.com/1/iperf3) and a quick check Iperf3 does not support the -u option for the server.

Also, Iperf3 requires a TCP control connection:
```
The initial TCP connection is used to exchange test parameters, control the start and end of the test,
and to exchange test results. This is sometimes referred to as the "control connection". The actual test
data is sent over a separate TCP connection, as a separate flow of UDP packets, or as an independent
SCTP connection, depending on what protocol was specified by the client.
```

Therefore, to establish a UDP stream the Iperf3 client requires an initial TCP connection before starting the UDP data flow. However, the platform & scenario configuration currently supports only Nodeports and services with a single protocol (TCP or UDP). For Iperf3 you need both TCP and UDP to be configured to use the same port. I believe this can be supported by the K8s but is not currently supported by our platform.

As a workaround, you can experiment with a custom user chart configuration for your Iperf3 application server to try to enable TCP and UDP on the same nodeport and service port. Otherwise, you can use Iperf instead of Iperf3 for your testing.

I hope this helps!
Regards!

**Reply by userH1213** [reply]
Hi @dilallkx,

  I see what you're saying now. You were referring to the syntax of the arguments, while I thought that you meant iperf3 as a tool. Well, regardless of the parameters/options that are sent, the server will run correctly because I'm using iperf3 python wrapper (V0.1.11).
![iperf3Wrapper](https://user-images.githubusercontent.com/79055649/196597513-cd6c0e0a-af51-4913-9cdf-429504a2a96d.PNG)

I'll definitely check your suggestions.
 Your help is appreciated
Thank you
